### ISSUE TITLE
Context is undefined in Cloud code beforeFind for queries using include

### ISSUE THREAD

**Issue by hej2010** [parse-community/parse-server#8759]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

I updated from Parse 6.2.0 to 6.3.0 and it appears to have changed how queries with `include(...)` are made.
The `req.context` is empty in `Parse.Cloud.beforeFind()` when using a query with `include()`.

For a query like:
```
const posts = await new Parse.Query("Post")
    .include("user") <- pointer to a _User
    ...
    .find({ context: {...}, sessionToken: ... });
```
in the old version the `beforeFind` was only called once and had the specified context object.

But in the latest (6.3.0) version the `beforeFind` is called twice, once for the query to `Post` (with the context object) and another time for the query to `_User` (with no context object and the query `{"where":{"objectId":{"$in":[objectId1, objectId2, ... ]}}}`). 

I use the `beforeFind` to only allow queries made from the cloud code for certain classes (by using the find context), but for queries with `include` it is no longer possible to check if the query was made from cloud code or from a client SDK.

### Steps to reproduce

Define beforeFind, like:
```
Parse.Cloud.beforeFind("Post", async req => {
    if (!req.master && !req.isGet) {
        if (!req.context) { throw "Empty") }
    }
});
Parse.Cloud.beforeFind(Parse.User, async req => {
    if (!req.master && !req.isGet) {
        if (!req.context) { throw "Empty") }
    }
});
```
and use a query like:
```
const posts = await new Parse.Query("Post")
    .include("user") <- pointer to a _User
    ...
    .find({ context: {...}, sessionToken: ... });
```

### Actual Outcome

The context in `beforeFind("Post", ...)` is the context object defined in the query.
The context in `beforeFind(Parse.User, ...)` (or any class that is included in a query) is undefined/empty.

### Expected Outcome

The context is defined in both `beforeFind` functions.

### Environment

Server
- Parse Server version: `6.3.0`

Database
- System (MongoDB or Postgres): `MongoDB`

**Comment by mtrezza** [comment]
Could you try the Parse Server versions in between 6.2 and 6.3, to we can pin down since when the issue was introduced? I assume it may be related to https://github.com/parse-community/parse-server/pull/8733, which has been released in 6.2.2.

**Comment by hej2010** [comment]
It works up to 6.2.1, but starting with 6.2.2 and up it has that issue.

**Comment by mtrezza** [comment]
Could you write a filing test for this in a PR? 

cc @Moumouls

**Comment by hej2010** [comment]
PR opened :)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.4.0-alpha.5](https://github.com/parse-community/parse-server/releases/tag/6.4.0-alpha.5)

**Comment by Moumouls** [comment]
thanks a lot @hej2010 !

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.5.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.5.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.5.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/6.5.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.5.0](https://github.com/parse-community/parse-server/releases/tag/6.5.0)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0](https://github.com/parse-community/parse-server/releases/tag/7.0.0)
